src/corehost/cli/fxr/fx_muxer.cpp

```diff
@@ -623,8 +623,17 @@ namespace
         }
 
         // Validate the current context is acceptable for this request (frameworks)
-        if (!fx_resolver_t::is_config_compatible_with_frameworks(app_config, existing_context->fx_versions_by_name))
-            return StatusCode::CoreHostIncompatibleConfig;
+        // Only validate for framework-dependent (i.e. non-empty frameworks)
+        // Self-contained apps don't contain information about frmeworks contained in the app, so there's nothing to validate against.
+        if (!existing_context->fx_versions_by_name.empty())
+        {
+            if (!fx_resolver_t::is_config_compatible_with_frameworks(app_config, existing_context->fx_versions_by_name))
+                return StatusCode::CoreHostIncompatibleConfig;
+        }
+        else
+        {
+            trace::verbose(_X("Skipped framework validation for loading a component in a self-contained app"));
```

Comment:
Do we want to specifically check for 3.0? Or at least assert? Or is this okay since it is only in the 3.0 branch?

Reply:
In theory we should check for >= 3.0, but since this feature only exists in 3.0, that check is moot. We can't check for == 3.0, because if happen to do nothing in 3.1 things would break on their own - bad.
So not having a version is basically the best solution for now.